File: mgmt.py
#python program to make and manage a teachers' student database management system

import csv
from sys import argv
from cs50 import get_int, get_string

def main(argv):

    if len(argv) != 2:
        print("Usage: python mgmt.py filename.csv")
        exit(1)

    file = open(argv[1], "r+")

    print("Enter the action you want to perform from the following menu: ")
    print("1. Write a new entry")
    print("2. Read the entries")
    action = get_int("Enter: ")

    if action == 1:
        enrNo = get_int("Enter enrNo: ")
        name = get_string("Enter name: ")
        branch = get_string("Enter branch: ")
        batch = get_string("Enter batch: ")
        marks = get_int("Enter marks: ")

        writer = csv.writer(file)
        writer.writerow((enrNo, name, branch, batch, marks))
    elif action == 2:
        print("Displaying the entries: ")
        reader = csv.reader(file, delimiter = ' ', quotechar = '|')
        for row in reader:
            print(row)

if __name__ == "__main__":
    main(argv)

File: mgmt.c
//Program to make and manage a Teachers' Students Database Management System

#include <stdio.h>
#include <stdlib.h>

void append(FILE *fileptr, int enrNo, char name[20], char branch[10], char batch[10], int marks);
void read(FILE *fileptr);

int main(int argc, char *argv[])
{
    if(argc != 2)
    {
        printf("Usage: ./mgmt filename.csv\n");
        exit(1);
    }

    FILE *fileptr = fopen(argv[1], "a+");

    int action;
    int enrNo, marks;
    char name[20], branch[10], batch[10];

    printf("Enter the action you want to perform form the following menu:\n");
    printf("1. Append a new entry\n");
    printf("2. Read entries\n");
    scanf("%d", &action);

    switch(action)
    {
        case 1: printf("Enter enrolment no, name, branch, batch and marks of student seperated by space:\n");
                scanf("%d %s %s %s %d", &enrNo, name, branch, batch, &marks);
                append(fileptr, enrNo, name, branch, batch, marks);
                break;
        case 2: printf("Displaying the entries:\n");
                read(fileptr);
    }

    fclose(fileptr);
}

// //Function to open a new file
// FILE *build()
// {
//     FILE *fileptr = fopen("management.csv", "a+");

//     return fileptr;
// }

//Append to a file
void append(FILE *fileptr, int enrNo, char name[20], char branch[10], char batch[10], int marks)
{
    fprintf(fileptr, "%d %s %s %s %d\n", enrNo, name, branch, batch, marks);
}
//Read the file
void read(FILE *fileptr)
{
    int end;
    int enrNo, marks;
    char name[20], branch[10], batch[10];

    while(1)
    {
        end = fscanf(fileptr, "%d %s %s %s %d", &enrNo, name, branch, batch, &marks);

        if(end < 0)
        {
            printf("Entries over\n");
            break;
        }
        printf("Enrollment No: %d\nName: %s\nBranch: %s\nBatch: %s\nMarks: %d\n", enrNo, name, branch, batch, marks);
    }
}